src/bsp/src/esp/src/Factory.cpp

```diff
@@ -0,0 +1,11 @@
+#include "bsp/Factory.h"
+#include "BSP.h"
+
+namespace BspFactory {
+
+std::shared_ptr<IBSP> getBSP() {
+    static BSP s_bsp;
+
+    return std::make_shared<BSP>(s_bsp);
```

Comment:
Don't return a shared pointer on a stack object, it can get destroyed and create undefined behavior, return a reference instead. And a reference makes it more clear that the user does not need to manage the memory

Reply:
The static BSP object is not a stack object. It is stored within the .bss section of the binary code, defined at build time (https://en.wikipedia.org/wiki/.bss). This is the use of having static objects contained within functions. All statics objects have their memory reserved, freeing the stack and heap of the program.

As such, there is no risk of deallocating this memory. I'd also argue that having a function that returns a shared_pointer is explicit enough that the user does not need to manage the memory and also explicitly tells the user he isn't the only owner of the reference.

Reply:
You are right that it's not a stack object, but it's still static memory and not dynamic memory. If your shared pointer reference falls to 0, the destructor will be called, what would happen then to your object?  

You could do this
https://stackoverflow.com/questions/20131877/how-do-you-make-stdshared-ptr-not-call-delete

But it totally defeats the purpose of a shared pointer. You could cast it instead, but there is no real gain from that

Reply:
I see. I am not sure we will have issues since no actions are made in the destructor in this situation, I will change it to return a reference and use the standard pointers as dependencies.

Reply:
Changed to reference.